src/server/scripts/Events/fireworks_show.cpp

```diff
@@ -0,0 +1,903 @@
+                started = true;
+            }
+
+            // Event is active
+            if ((localTm.tm_min >= 0 && localTm.tm_sec >= 1 && localTm.tm_min <= 9 && localTm.tm_sec <= 59) && !started && IsHolidayActive(HOLIDAY_FIREWORKS_SPECTACULAR))
+            {
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                started = true;
+            }
+
+            // Stop
+            if ((localTm.tm_min == 10 && localTm.tm_sec == 0) || (localTm.tm_min == 10 && localTm.tm_sec == 0 && localTm.tm_hour == 0 && started == true))
+            {
+                started = false;
+                _events.ScheduleEvent(EVENT_CHEER, Seconds(1));
+                _events.CancelEvent(EVENT_FIRE);
+            }
+
+            // New Year (Only!) - One more big bang!
+            if ((localTm.tm_min == 10 && localTm.tm_sec == 30 && localTm.tm_hour == 0) && IsEventActive(GAME_EVENT_NEW_YEAR) && big == true)
+            {
+                big = false;
+                _events.ScheduleEvent(EVENT_CHEER, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+                _events.ScheduleEvent(EVENT_FIRE, Seconds(1));
+            }
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_CHEER:
+                    {
+                        go->PlayDistanceSound(CheerPicker());
+                        break;
+                    }
+                    case EVENT_FIRE:
+                    {
+                        auto it = pos.find(go->GetZoneId());
+                        if (it != pos.end())
+                        {
+                            std::vector<Position> const& zoneFireworks = it->second;
+                            uint32 rndpos = urand(0, zoneFireworks.size());
+                            float rndrot = frand(-1.0000000f, 1.0000000f);
+                            float rndrot2 = frand(-1.0000000f, 1.0000000f);
+
+                            if (big)
+                            {
+                                if (GameObject* firework = go->SummonGameObject(FireworksBIGOnlyPicker(), zoneFireworks[rndpos], G3D::Quat(0.f, 0.f, rndrot, rndrot2), 300))
+                                {
+                                    firework->setActive(true);
+                                    firework->SetRespawnTime(0);
+                                    firework->Delete();
```

Comment:
but eitherway you are preventing the grid unload from the event activation till the next server restart, not very neat xd
like I said just spawn again the gameobjects (if its necessary) when the main gob spawns (aka when this ai loads)